### ISSUE TITLE
Behavior for missing arguments

### ISSUE THREAD

**Issue by jimhester** [gaborcsardi/argufy#13]
``` r
f1 <- function(x = ? is.character) {
  if (missing(x)) {
    "missing"
  } else {
    x
  }
}
```

Currently `argufy(f1)()` throws

```
Error in stopifnot(is.character(x)) :
  argument "x" is missing, with no default
```

I am not sure if this is should be the behavior or not, intuitively I was only expecting arguments to be checked if they are supplied.  Not sure how tricky the fix for this would be, maybe just modify the test to `stopifnot(!missing(x) && is.character(x))`

**Comment by gaborcsardi** [comment]
YEah, I think we need `!missing()`, the current behavior is surprising, and also limiting.

**Comment by gaborcsardi** [comment]
I guess we need to wrap everything into `if (!missing)`.

**Comment by jimhester** [comment]
Fixed by 3aed4de1d507efde7e6238f880bdedde35d5f075
